base/test/sound_record_tests.cpp

```diff
@@ -12,11 +13,30 @@
 #include "Module.h"
 #include<iostream>
 #include<fstream>
+#include<vector>
 
 BOOST_AUTO_TEST_SUITE(sound_record_tests)
 
+struct FileCleaner {
```

Comment:
Can't we reuse the last version of this function that was written earlier... it is never a good idea to copy paste code...
I suggest we move this "struct FileCleaner" int to test_utils and reuse everywhere.  
 Once we do this properly it will become an example for other tests. 
On the contrary if we do it wrong then the wrong habit will get copy pasted to all tesst @mraduldubey 

Rest is all good.